tests/test_packageAlerts.py

```diff
@@ -95,9 +110,46 @@ def _roundTripThroughApdb(objects, sources, forcedSources, dateTime):
     return (diaObjects, diaSources, diaForcedSources)
 
 
+def mock_alert(alert_id):
+    """Generate a minimal mock alert.
+    """
+    return {
+        "alertId": alert_id,
+        "diaSource": {
+            # Below are all the required fields containing random values.
+            "midpointMjdTai": 5,
+            "diaSourceId": 4,
+            "ccdVisitId": 2,
+            "band": 'g',
+            "ra": 12.5,
+            "dec": -16.9,
+            "x": 15.7,
+            "y": 89.8,
+            "apFlux": 54.85,
+            "apFluxErr": 70.0,
+            "snr": 6.7,
+            "psfFlux": 700.0,
+            "psfFluxErr": 90.0,
+            "flags": 0,
+        }
+    }
+
+
 class TestPackageAlerts(lsst.utils.tests.TestCase):
+    kafka_enabled = "confluent_kafka" in sys.modules
+
+    def __init__(self, *args, **kwargs):
+        TestPackageAlerts.kafka_enabled = "confluent_kafka" in sys.modules
+        _log.debug('TestPackageAlerts: kafka_enabled={}'.format(self.kafka_enabled))
+        super(TestPackageAlerts, self).__init__(*args, **kwargs)
```

Comment:
I don't think you want this at all: you want to set a global in the `except ModuleNotFoundError as e:` above and decorate either the whole class (if all tests in the class need kafka) or the kafka-specific methods with `@unittest.skipIf`.

https://docs.python.org/3/library/unittest.html#unittest.skipIf

Reply:
Removed and swapped to Eli's method of setting confluent_kafka to None.